### PR TITLE
Savva Dzhumaliev Tasks 41-50

### PR DESCRIPTION
**Pull request by EasileleM**

### CODE REVIEW

File: tasks/41_sort-by-bits/sort-by-bits.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    if (a & (1 << i)) {
+      result++;
+    }
+  }
+  return result;
+}
+
+export function sortByBits(arr) {
+  return arr.sort((a, b) => {
+    let bitsA = countBits(a);
```
Comment (vovcyan) [inline tasks/41_sort-by-bits/sort-by-bits.js]: const

File: tasks/42_first-n-smallest/first-n-smallest.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export function firstNSmallest(arr, n) {
+  let arrCloneRev = Array(...arr).reverse();
```
Comment (vovcyan) [inline tasks/42_first-n-smallest/first-n-smallest.js]: const

File: tasks/45_equal-sides/equal-sides.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export function findEqualIndex(arr) {
+  // 0 in args of reduce is necessary to avoid exception
+  // which would appear in case where arr is an empty array
+  let sumRight = arr.reduce((sum, value) => sum + value, 0);
```
Comment (vovcyan) [inline tasks/45_equal-sides/equal-sides.js:4]: const

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/45_equal-sides/equal-sides.js (hunk 3 above)
Comment (EasileleM) [reply tasks/45_equal-sides/equal-sides.js:4]: This variable changes through execution

File: tasks/47_string-expansion/string-expansion.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export function stringExpansion(str) {
```
Comment (akimovaa) [inline tasks/47_string-expansion/string-expansion.js:1]: You can find solution with only one loop, w/o too many operations.

**Review by akimovaa — CHANGES_REQUESTED** [review]

**Review by akimovaa — APPROVED** [review]

File: tasks/41_sort-by-bits/sort-by-bits.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+//       .toString(2)
+//       .match(/[1]/g) || [])
+//       .length;
+// }
+
+// Bitwise version, faster than previous
+function countBits(a) {
+  let result = 0;
+  for (let i = 0; i < 32; i++) {
+    if (a & (1 << i)) {
```
Comment (KalachevDev) [inline tasks/41_sort-by-bits/sort-by-bits.js:13]: oh my lord, it is computer science stuff here!

**Review by KalachevDev — APPROVED** [review]

File: tasks/44_time-simulation/time-simulation.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  }
+  get() {
+    return this.time;
+  }
+  setSpeed(newSpeed) {
+    this.speed = newSpeed;
+  }
+  update(currentRealTime) {
+    if (currentRealTime < this.realTime) {
+      throw String('Time mustn\'t go backwords');
```
Comment (KalachevDev) [inline tasks/44_time-simulation/time-simulation.js:15]: Backw**a**rds

**Review by KalachevDev — APPROVED** [review]

File: tasks/44_time-simulation/time-simulation.js (hunk 6 above)
Comment (EasileleM) [reply tasks/44_time-simulation/time-simulation.js:15]: Ty xD

**Pull request merged**